tex/generic/pgf/utilities/pgfkeys.code.tex

```diff
@@ -180,7 +187,7 @@
 %
 % The length is \pgfkeysvalue{/tikz/length}.
 
-\def\pgfkeysvalueof#1{\csname pgfk@#1\endcsname}
+\long\def\pgfkeysvalueof#1{\csname\pgfkeys@ifcsname{pgfk@#1}{pgfk@#1}{relax}\endcsname}
```

Comment:
This is not feasible. It changes the number of expansion steps required to get to the result. A `\csname` being a primitive expands in a single step but if we put the check outside if would be `\pgfkeys@ifcsname` -> `\ifcsname` -> `\pgfkeys@firstoftwo` -> `\csname` which is suddenly four expansions. If clients rely on the number of `\expandafter`s necessary to get the value out this will be breaking for them.

I actually think I also have to partially revert my changes to `\pgfkeysifdefined`. It also requires one more step now.

For the non-expandable ones like `\pgfkeysgetvalue`, etc. this doesn't matter. We can turn the check inside out if that increases readability.

Reply:
> It changes the number of expansion steps required to get to the result.

Ah I never thought from this aspect before (though it's possible to simulate by `\pgfkeys@expanded` but that will also complicate the implementation). Keeping compatibility for undocumented behavior is really a tough thing.

Reply:
`\pgfkeys@expanded` will not solve the problem because easily becaues we don't want to expand what's stored in the key as well. :wink: These types of mistakes also happen to me quite frequently.

Reply:
I had `\pgfkeys@unexpanded\expandafter{\csname pgfk@#1\endcsname}` in mind when commented but didn't typed it out. That's why I said "... will also complicate the implementation". Perhaps it's English that prohibits me from typing out everything I thought, or it's part of my bad habit. sorry

Reply:
Turning the test inside out for the non-expandable ones might be faster in the undefined case since TeX doesn't have to build `\relax` that way but instead in can be directly provided.